qemu/tests/hotplug_mem_negative.py

```diff
@@ -60,8 +61,19 @@ def check_msg(keywords, msg):
             test.fail("No valid keywords were found in the qemu prompt message")
 
     if params["size_mem"] == "<overcommit>":
+        ipa_limit_check = params.get('ipa_limit_check')
         overcommit_mem = normalize_data_size("%sK" % (memory.memtotal() * 2), "G")
         params["size_mem"] = "%sG" % round(float(overcommit_mem))
+        if ipa_limit_check:
+            system_init_mem = int(params["system_init_mem"])
+            slots_mem = int(params["slots_mem"])
+            extend_mem_region = int(params["extend_mem_region"])
+            ipa_limit = process.run(ipa_limit_check, shell=True).stdout.decode().strip() or 40
+            ipa_limit_size = 1 << int(ipa_limit)
+            ipa_limit_size = int(normalize_data_size("%sB" % ipa_limit_size, "G"))
+            params["maxmem_mem"] = "%dG" % (ipa_limit_size - system_init_mem -
+                                            (1 * slots_mem) - extend_mem_region)
```

Comment:
Would better compare the size between this and `maxmem_mem` in cfg, choose the small one.

Reply:
good points! updating

Reply:
corrected